pytorch_lightning/callbacks/progress/rich_progress.py

```diff
@@ -227,10 +228,15 @@ def __init__(
         leave: bool = False,
         theme: RichProgressBarTheme = RichProgressBarTheme(),
     ) -> None:
-        if not _RICH_AVAILABLE:
+        if not _module_available("rich"):
             raise ModuleNotFoundError(
                 "`RichProgressBar` requires `rich` to be installed. Install it by running `pip install -U rich`."
             )
+        if not _RICH_GREATER_EQUAL_10_2_2:
+            raise MisconfigurationException(
+                "`RichProgressBar` requires `rich` >= 10.2.2. Install it by running `pip install -U rich`."
+            )
```

Comment:
It looks to me that this message can directly replace the above message and be used for both the initial installation notice as well as for letting the user know they need to upgrade.

Reply:
Yes, this is what I meant with my comment above

Reply:
Updated!